commit 8ee86aa826f74b2a0446dfccb014534d853525f0
Author: sjorge <sjorge@blackdot.be>
Date:   2019-09-29T18:22:16+02:00 (8 days ago)
    
    OS-8003 Do not destroy external delegated dataset
    
    Do not recursively destroy dataset that are external to the zonepath.
    This makes juninstall_delegated_dataset obsolete. And brings joyent
     and joyent-minimal brand inline with the lx brand.

diff --git a/usr/src/lib/brand/jcommon/libhooks.ksh b/usr/src/lib/brand/jcommon/libhooks.ksh
index 346356519c..913d81899b 100644
--- a/usr/src/lib/brand/jcommon/libhooks.ksh
+++ b/usr/src/lib/brand/jcommon/libhooks.ksh
@@ -16,7 +16,7 @@
 #
 
 #
-# This file contains various hooks that are used by nore than a single
+# This file contains various hooks that are used by more than a single
 # brand. This file should be included by the brand-specific files.
 #
 
@@ -84,13 +84,3 @@ jattach_zone_final_setup()
 
 	touch $ZROOT/var/log/courier.log
 }
-
-function juninstall_delegated_dataset
-{
-	# Now destroy any delegated datasets. Redirect to /dev/null in case they
-	# were already destroyed when we removed the zonepath dataset.
-	DD=`zonecfg -z $ZONENAME info dataset | nawk '{if ($1 == "name:") print $2}'`
-	for i in $DD; do
-		zfs destroy -rF $i >/dev/null 2>&1
-	done
-}
diff --git a/usr/src/lib/brand/joyent-minimal/zone/juninstall b/usr/src/lib/brand/joyent-minimal/zone/juninstall
index 09051a17b6..136044ad9e 100755
--- a/usr/src/lib/brand/joyent-minimal/zone/juninstall
+++ b/usr/src/lib/brand/joyent-minimal/zone/juninstall
@@ -15,11 +15,9 @@
 # Copyright (c) 2014 Joyent, Inc.  All rights reserved.
 #
 
-. /usr/lib/brand/jcommon/libhooks.ksh
-
 function jcommon_uninstall_hook
 {
-	juninstall_delegated_dataset
+	:
 }
 
 . /usr/lib/brand/jcommon/cuninstall
diff --git a/usr/src/lib/brand/joyent/zone/juninstall b/usr/src/lib/brand/joyent/zone/juninstall
index 09051a17b6..136044ad9e 100755
--- a/usr/src/lib/brand/joyent/zone/juninstall
+++ b/usr/src/lib/brand/joyent/zone/juninstall
@@ -15,11 +15,9 @@
 # Copyright (c) 2014 Joyent, Inc.  All rights reserved.
 #
 
-. /usr/lib/brand/jcommon/libhooks.ksh
-
 function jcommon_uninstall_hook
 {
-	juninstall_delegated_dataset
+	:
 }
 
 . /usr/lib/brand/jcommon/cuninstall
